pandas/io/formats/printing.py

```diff
@@ -528,3 +528,10 @@ def format_object_attrs(
     if len(obj) > max_seq_items:
         attrs.append(("length", len(obj)))
     return attrs
+
+
+class PrettyDict(dict):
```

Comment:
with mypy -disallow-any-generic, see #30539, this gives `pandas\io\formats\printing.py:533: error: Implicit generic "Any". Use "typing.Dict" and specify generic parameters`
```suggestion
class PrettyDict(Dict[_KT, _VT]):
```

and add
```python
_KT = TypeVar("_KT")
_VT = TypeVar("_VT")
```